src/main/groovy/nebula/plugin/dependencylock/tasks/DiffLockTask.groovy

```diff
@@ -26,20 +29,32 @@ class DiffLockTask extends AbstractLockTask {
     File outputDir = new File(project.buildDir, "dependency-lock")
 
     @OutputFile
-    File diffFile = new File(outputDir, "lockdiff.txt")
+    File diffFile = new File(outputDir, "lockdiff.${this.diffFileExtension()}")
+
+    private String diffFileExtension() {
+        DependencyLockingFeatureFlags.isPathAwareDependencyDiffEnabled() ? "json" : "txt"
+    }
 
     @TaskAction
     def diffLocks() {
         ConfigurationsSet existingLock = readLocks(existingLockFile)
         ConfigurationsSet newLock = readLocks(updatedLockFile)
-        if (newLock.isEmpty()) {
+        if (DependencyLockingFeatureFlags.isPathAwareDependencyDiffEnabled()) {
+            Map<String, List<DependencyDiff>> diffByConfiguration = new DependenciesComparison().performDiffByConfiguration(existingLock, newLock)
+            DiffReportGenerator generator = Class.forName("nebula.plugin.dependencylock.diff.PathAwareDiffReportGenerator").newInstance() as DiffReportGenerator
```

Comment:
@DanielThomas Is there a better way to call kotlin code from groovy? As far as I understand groovy is compiled first and then kotlin

Reply:
You can order Groovy/Kotlin compilation to work the way that's most convenient for interop. Usually I make the Groovy classes visible to Kotlin, and use reflection from Groovy to Kotlin. There's an example configuration in the internal Nebula repository.